metadata-ingestion/src/datahub/upgrade/upgrade.py

```diff
@@ -55,11 +55,19 @@ async def get_client_version_stats():
         async with session.get(pypi_url) as resp:
             response_json = await resp.json()
             try:
-                releases = response_json.get("releases", [])
-                sorted_releases = sorted(releases.keys(), key=lambda x: Version(x))
-                latest_cli_release_string = [
-                    x for x in sorted_releases if "rc" not in x
-                ][-1]
+                releases = response_json.get("releases", {})
+                filtered_releases = {
+                    version: release_files
+                    for version, release_files in releases.items()
+                    if not all(
```

Comment:
`not all` or `not any` ? 🤔

Reply:
We have two files per release - a .tar.gz and .whl file. If in the future we add architecture-specific .whl files, we'll have multiple. We should only really consider a release yanked if every file is yanked - if we yank only a specific file due to a packaging issue, the release is still mostly valid